aws/s3.go

```diff
@@ -88,55 +97,131 @@ func getAllS3Buckets(awsSession *session.Session, excludeAfter time.Time, bucket
 	var bucketNamesPerRegion = make(map[string][]*string)
 	var regionClients = make(map[string]*s3.S3)
 
-	for _, bucket := range output.Buckets {
-		logging.Logger.Debugf("Checking - Bucket %s", *bucket.Name)
+	for _, region := range targetRegions {
+		logging.Logger.Debugf("S3 - creating session - region %s", region)
+		awsSession, err := session.NewSession(&awsgo.Config{
+			Region: awsgo.String(region)},
+		)
+		if err != nil {
+			return bucketNamesPerRegion, errors.WithStackTrace(err)
+		}
+		regionClients[region] = s3.New(awsSession)
+	}
+
+	totalBuckets := len(output.Buckets)
+	if totalBuckets == 0 {
+		return bucketNamesPerRegion, nil
+	}
+
+	if batchSize <= 0 {
+		return nil, fmt.Errorf("Invalid batchsize - %d - should be > 0", batchSize)
+	}
+
+	totalBatches := totalBuckets / batchSize
+	if totalBuckets%batchSize != 0 {
+		totalBatches = totalBatches + 1
+	}
+	batchCount := 1
+
+	// Batch the get operation
+	for i := 0; i < totalBuckets; i += batchSize {
+		j := i + batchSize
```

Comment:
`i` and `j` are hard to follow here. Perhaps `batchStart` and `batchEnd`?

Reply:
Yes. Done.